redaxo/src/addons/mediapool/functions/function_rex_mediapool.php

```diff
@@ -598,21 +598,22 @@ function rex_mediapool_isAllowedMediaType($filename, array $args = [])
  *
  * @return bool
  */
-function rex_mediapool_isAllowedMimeType($path)
+function rex_mediapool_isAllowedMimeType($file_path, $file_name)
```

Comment:
der 1. parameter sollte vllt darauf hinweisen dass das der tmp-path ist, z.B. `$tmp_file_path` o.ä.

Reply:
sollten wir den zweiten optional machen? (damit, falls man die methode mit einem normalen (nicht-tmp)-Pfad nutzt, man den nicht doppelt übergeben muss?
Oder lieber explizit den zweiten immer erwarten?

Reply:
hmm nochmal drüber nachgedacht... sollte die funktion nicht einfach nur 1 parameter haben und dieser soll der vollständige pfad sein zum fertigen file (nicht dem temp pfad)?

Reply:
Geprüft wird ja, bevor die Upload-Tempdatei in den MP geschoben wird. Und diese Tempdatei (aus `$_FILES`) hat keine Dateiendung.

Reply:
anders gesagt: man würde erwarten dass man der funktionen einen dateipfad zur verfügung stellt und das system sagt mir für genau diesen dateipfad obs passt oder nicht.

ob das jetzt temp dateien sind oder nicht, sollte der funktion egal sein....

Reply:
Ja, so wäre es schön. Aber dann müssen wir die Uploaddatei ja erst noch selbst an einen anderen temporären Ort schieben. An beiden Stellen unten im PR-Diff, wo die Funktion verwendet wird.

Diese Funktion hier prüft nun mal sowohl die Dateiextension anhand des Dateinamens, als auch den Mimetype anhand des Dateiinhalts.

Reply:
wenn wir datei vorher nicht kopieren/moven wollen, dann muss es wohl so bleiben wie es ist, und entsprechend phpdoc her für das etwas "merkwürdige" verhalten.

ausserdem muss der 2. parameter optional sein, da es sonst ein BC break ist.
jeder consumer dieser methode, der vllt kein upload vorher gemacht hat, geht ja bis heute davon aus, dass diese methode nur 1 parameter hat.

Reply:
ich würde einfach den zweiten Parameter optional machen, sodass falls dieser nicht gesetzt wurde, der Dateiname/extension anhand des 1. Parameter gewählt wird.